src/main.rs

```diff
@@ -601,6 +601,20 @@ mod unit_tests {
                 }"#,
                 false,
             ),
+            (
+                vec!["cloud-hypervisor", "--memory", "size=1G,hotplug_size=1G"],
+                r#"{
+                    "memory": {"size": 1073741824, "hotplug_method": "ACPI", "hotplug_size": 1073741824}
+                }"#,
+                true,
+            ),
+            (
+                vec!["cloud-hypervisor", "--memory", "size=1G,hotplug_method=virtio-mem,hotplug_size=1G"],
+                r#"{
+                    "memory": {"size": 1073741824, "hotplug_method": "VirtioMem", "hotplug_size": 1073741824}
+                }"#,
+                true,
+            ),
```

Comment:
So this is a unit test, that's nice to have, but we also need an integration test (they're located at `tests/integration.rs`).

Reply:
The virtio-mem is still not upstream and I think it is not support by the current kernel that run the  integration test.
That is why I didn't add test for it.

Reply:
We need to update the custom kernel branch so that it includes the patches. We're already doing this for virtio-iommu and virtio-fs.
I can work on creating a new kernel branch with the virtio-mem patches added. In the meantime, please create the integration test.

Reply:
Done with integration test.